File: frontend/src/api/system.ts
import request from './request'

export interface SystemLog {
  id: number
  level: string
  module: string
  action: string
  content: string
  user_id: number
  ip: string
  created_at: string
}

export interface Menu {
  id?: number
  name: string
  path: string
  icon: string
  parent_id: number
  sort: number
  visible: boolean
  roles: string
  created_at?: string
  updated_at?: string
}

export interface User {
  id?: number
  username: string
  nickname: string
  role: string
  password?: string
  created_at?: string
  updated_at?: string
}

export const getSystemLogs = (params: any) => {
  return request.get<any, { list: SystemLog[]; total: number; page: number }>('/system/logs', { params })
}

export const createSystemLog = (data: Partial<SystemLog>) => {
  return request.post('/system/logs', data)
}

export const getMenus = () => {
  return request.get<any, Menu[]>('/system/menus')
}

export const createMenu = (data: Menu) => {
  return request.post<any, Menu>('/system/menus', data)
}

export const updateMenu = (id: number, data: Menu) => {
  return request.put<any, Menu>(`/system/menus/${id}`, data)
}

export const deleteMenu = (id: number) => {
  return request.delete(`/system/menus/${id}`)
}

export const getUsers = () => {
  return request.get<any, User[]>('/system/users')
}

export const createUser = (data: User) => {
  return request.post<any, User>('/system/users', data)
}

export const updateUser = (id: number, data: Partial<User>) => {
  return request.put<any, User>(`/system/users/${id}`, data)
}

export const deleteUser = (id: number) => {
  return request.delete(`/system/users/${id}`)
}

File: internal/app/model/system.go
package model

import "time"

type SystemLog struct {
	ID        uint      `gorm:"primarykey" json:"id"`
	Module    string    `json:"module"`
	Action    string    `json:"action"`
	Content   string    `gorm:"type:text" json:"content"`
	UserID    uint      `json:"user_id"`
	Username  string    `json:"username"`
	IP        string    `json:"ip"`
	CreatedAt time.Time `json:"created_at"`
}

type Menu struct {
	ID        uint      `gorm:"primarykey" json:"id"`
	Name      string    `gorm:"not null" json:"name"`
	Path      string    `json:"path"`
	Icon      string    `json:"icon"`
	ParentID  *uint     `json:"parent_id"`
	Sort      int       `gorm:"default:0" json:"sort"`
	Visible   bool      `gorm:"default:true" json:"visible"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}
